File: 169_majority_element/majorityElementHash.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> hash = new HashMap<>();

        int n = nums.length / 2;

        for (int i : nums) {
            int value = hash.getOrDefault(i, 0) + 1;
            hash.put(i, value);
        }

        for (Map.Entry<Integer, Integer> entry : hash.entrySet()) {
            Integer key = entry.getKey();
            Integer value = entry.getValue();

            if (value > n) {
                return key;
            }
        }

        return 0;
    }
}

File: 169_majority_element/majorityElementHash.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hash = {}

        for i in nums:
            if i in hash:
                hash[i] = hash[i] + 1
            else:
                hash[i] = 1

        n = len(nums) // 2

        for key, val in hash.items():
            if val > n:
                return key 